"Class XXX used before its declaration" breaking change in 2.3.0

From https://github.com/Microsoft/TypeScript/issues/15102#issuecomment-302877013

This is still showing errors if the class and its use are in separate files.

**a.ts**
```ts
let foo: typeof C;
```

**b.ts**
```ts
class C {}
```

**tsconfig.json**
```json
{
  "compilerOptions": {
      "outFile": "foo"
  }
}
```

Shows: `src/a.ts(1,17): error TS2449: Class 'C' used before its declaration.`

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -735,6 +735,7 @@ namespace ts {
             if (declarationFile !== useFile) {
                 if ((modulekind && (declarationFile.externalModuleIndicator || useFile.externalModuleIndicator)) ||
                     (!compilerOptions.outFile && !compilerOptions.out) ||
+                    isInTypeQuery(usage) ||
                     isInAmbientContext(declaration)) {
                     // nodes are in different files and order cannot be determined
                     return true;

--- a/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.js
+++ b/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.js
@@ -1,0 +1,15 @@
+//// [tests/cases/compiler/blockScopedClassDeclarationAcrossFiles.ts] ////
+
+//// [c.ts]
+let foo: typeof C;
+//// [b.ts]
+class C { }
+
+
+//// [foo.js]
+var foo;
+var C = (function () {
+    function C() {
+    }
+    return C;
+}());

--- a/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.symbols
+++ b/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.symbols
@@ -1,0 +1,9 @@
+=== tests/cases/compiler/c.ts ===
+let foo: typeof C;
+>foo : Symbol(foo, Decl(c.ts, 0, 3))
+>C : Symbol(C, Decl(b.ts, 0, 0))
+
+=== tests/cases/compiler/b.ts ===
+class C { }
+>C : Symbol(C, Decl(b.ts, 0, 0))
+

--- a/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.types
+++ b/tests/baselines/reference/blockScopedClassDeclarationAcrossFiles.types
@@ -1,0 +1,9 @@
+=== tests/cases/compiler/c.ts ===
+let foo: typeof C;
+>foo : typeof C
+>C : typeof C
+
+=== tests/cases/compiler/b.ts ===
+class C { }
+>C : C
+

--- a/tests/cases/compiler/blockScopedClassDeclarationAcrossFiles.ts
+++ b/tests/cases/compiler/blockScopedClassDeclarationAcrossFiles.ts
@@ -1,0 +1,5 @@
+// @outFile: foo.js
+// @Filename: c.ts
+let foo: typeof C;
+// @Filename: b.ts
+class C { }

